test/MeasureBundleHelpers.test.ts

```diff
@@ -283,4 +285,48 @@ describe('MeasureBundleHelpers', () => {
       );
     });
   });
+
+  describe('addValueSetsToMeasureBundle', () => {
+    test('throws an error if no API key is provided for retrieving the ValueSet resource(s)', async () => {
+      const measureBundle: fhir4.Bundle = getJSONFixture('EXM130-7.3.000-bundle-nocodes-missingVS.json');
+
+      try {
+        await MeasureBundleHelpers.addValueSetsToMeasureBundle(measureBundle);
+      } catch (e) {
+        expect(e.message).toEqual(
+          'Missing the following valuesets: http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.464.1003.101.12.1016, and no API key was provided to resolve them'
+        );
+      }
+    });
+
+    test('returns original measure bundle if measure bundle is not missing any ValueSet resources', async () => {
+      const measureBundle: fhir4.Bundle = getJSONFixture('EXM130-7.3.000-bundle-nocodes.json');
+      const returnedBundle = await MeasureBundleHelpers.addValueSetsToMeasureBundle(measureBundle, 'an_api_key');
+      expect(returnedBundle).toEqual(measureBundle);
+    });
+
+    test('returns new bundle with added ValueSet resource when measure bundle is missing one ValueSet resource', async () => {
+      // measure bundle with one missing ValueSet
+      const measureBundle = getJSONFixture('EXM130-7.3.000-bundle-nocodes-missingVS.json');
+      // missing ValueSet resource
+      const missingVS = getJSONFixture('valuesets/example-missing-EXM130-vs.json');
+
+      const vsrSpy = jest
+        .spyOn(ValueSetResolver.prototype, 'getExpansionForValuesetUrls')
+        .mockImplementation(async () => {
+          {
+            return [[missingVS], []];
+          }
+        });
+      const returnedBundle = await MeasureBundleHelpers.addValueSetsToMeasureBundle(measureBundle, 'an_api_key');
+
+      expect(vsrSpy).toHaveBeenCalledWith(getMissingDependentValuesets(measureBundle));
+      expect(returnedBundle.entry?.length).toEqual(measureBundle.entry?.length);
+      expect(returnedBundle.entry?.slice(returnedBundle.entry?.length - 1)[0]).toEqual(missingVS);
+    });
+
```

Comment:
Good idea, I can try to add that :) I originally thought the ValueSetResolver.test.ts error test would be enough, but I agree it might be useful to add a unit test here as well to signify that the new bundle doesn't get made if errors come back from the ValueSetResolver